flink-runtime/src/test/java/org/apache/flink/runtime/io/network/partition/hybrid/tiered/netty/TieredStoreResultSubpartitionViewTest.java

```diff
@@ -0,0 +1,205 @@
+        checkBufferAndBacklog(tieredStoreResultSubpartitionView.getNextBuffer(), 0);
+        tieredStoreResultSubpartitionView.notifyRequiredSegmentId(1);
+        assertThat(availabilityListener).isDone();
+        checkBufferAndBacklog(tieredStoreResultSubpartitionView.getNextBuffer(), 1);
+        checkBufferAndBacklog(tieredStoreResultSubpartitionView.getNextBuffer(), 0);
+        assertThat(tieredStoreResultSubpartitionView.getNextBuffer()).isNull();
+    }
+
+    @Test
+    void testGetNextBufferFailed() {
+        Throwable expectedError = new IOException();
+        nettyPayloadQueues = createNettyPayloadQueuesWithError(expectedError);
+        tieredStoreResultSubpartitionView =
+                new TieredStoreResultSubpartitionView(
+                        createBufferAvailabilityListener(availabilityListener),
+                        nettyPayloadQueues,
+                        createNettyConnectionIds(),
+                        createNettyServiceProducers(connectionBrokenConsumers));
+        assertThatThrownBy(tieredStoreResultSubpartitionView::getNextBuffer)
+                .hasCause(expectedError);
+        assertThat(connectionBrokenConsumers.get(0).isDone()).isTrue();
+    }
+
+    @Test
+    void testGetAvailabilityAndBacklog() {
+        ResultSubpartitionView.AvailabilityWithBacklog availabilityAndBacklog1 =
+                tieredStoreResultSubpartitionView.getAvailabilityAndBacklog(0);
+        assertThat(availabilityAndBacklog1.getBacklog()).isEqualTo(3);
+        assertThat(availabilityAndBacklog1.isAvailable()).isEqualTo(false);
+        ResultSubpartitionView.AvailabilityWithBacklog availabilityAndBacklog2 =
+                tieredStoreResultSubpartitionView.getAvailabilityAndBacklog(2);
+        assertThat(availabilityAndBacklog2.getBacklog()).isEqualTo(3);
+        assertThat(availabilityAndBacklog2.isAvailable()).isEqualTo(true);
+    }
+
+    @Test
+    void testNotifyRequiredSegmentId() {
+        tieredStoreResultSubpartitionView.notifyRequiredSegmentId(1);
+        assertThat(availabilityListener.isDone()).isTrue();
+    }
+
+    @Test
+    void testReleaseAllResources() throws IOException {
+        tieredStoreResultSubpartitionView.releaseAllResources();
+        assertThat(nettyPayloadQueues.get(0)).hasSize(0);
+        assertThat(nettyPayloadQueues.get(1)).hasSize(0);
+        assertThat(connectionBrokenConsumers.get(0).isDone()).isTrue();
+        assertThat(connectionBrokenConsumers.get(1).isDone()).isTrue();
+        assertThat(tieredStoreResultSubpartitionView.isReleased()).isTrue();
+    }
+
+    @Test
+    void testGetNumberOfQueuedBuffers() {
+        assertThat(tieredStoreResultSubpartitionView.getNumberOfQueuedBuffers()).isEqualTo(3);
+        assertThat(tieredStoreResultSubpartitionView.unsynchronizedGetNumberOfQueuedBuffers())
+                .isEqualTo(3);
+    }
+
+    private void checkBufferAndBacklog(BufferAndBacklog bufferAndBacklog, int backlog) {
```

Comment:
```suggestion
    private static void checkBufferAndBacklog(BufferAndBacklog bufferAndBacklog, int backlog) {
```

I checked all the private methods in the test classes, they should be static.